### PR TITLE
Adding link for Angular Testing Succinctly

### PR DESCRIPTION
**Pull request by ManviGoel26**
## What does this PR do?
Add resource(s) 

### Description
It is a free book about Angular Testing. 

### Why is this valuable (or not)?
In Angular Testing Succinctly, author Joseph D. Booth provides a quick introduction to different tools and approaches to testing Angular projects to make them higher quality from the start and easier to maintain down the road. Topics covered include lint checking, test-driven development, unit testing, and end-to-end testing.

### How do we know it's really free?
The book and GitHub Repo are available online for free.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book. 

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1164,6 +1164,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 * [Angular 1 Style Guide](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md) - John Papa (HTML)
 * [Angular for the jQuery developer](http://www.ng-newsletter.com.s3-website-us-east-1.amazonaws.com/posts/angular-for-the-jquery-developer.html) - ng-newsletter, Ari Lerner, David Wu, Nate Murray (HTML)
+* [Angular Testing Succinctly](https://www.syncfusion.com/succinctly-free-ebooks/angular-testing-succinctly) - Joseph D. Booth (HTML, PDF, EPUB, [Companion Repo and example codes](https://github.com/SyncfusionSuccinctlyE-Books/Angular-Testing-Succinctly)) (email address required for EPUB and PDF)
```
Comment (shrn01) [inline books/free-programming-books-langs.md:1167]: Companion repo is mentioned in the book?

Comment (ManviGoel26) [reply books/free-programming-books-langs.md:1167]: Yes, it is mentioned in the GitHub Repo and the book's official page as source code.

Comment (shrn01) [reply books/free-programming-books-langs.md:1167]: Then you can remove it

Comment (shrn01) [inline books/free-programming-books-langs.md:1167]:
```suggestion
* [Angular Testing Succinctly](https://www.syncfusion.com/succinctly-free-ebooks/angular-testing-succinctly) - Joseph D. Booth (HTML, PDF, EPUB) (email address required for EPUB and PDF)
```

Comment (ManviGoel26) [reply books/free-programming-books-langs.md]: Ok, I have removed it.

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -1164,6 +1164,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 
 * [Angular 1 Style Guide](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md) - John Papa (HTML)
 * [Angular for the jQuery developer](http://www.ng-newsletter.com.s3-website-us-east-1.amazonaws.com/posts/angular-for-the-jquery-developer.html) - ng-newsletter, Ari Lerner, David Wu, Nate Murray (HTML)
+* [Angular Testing Succinctly](https://www.syncfusion.com/succinctly-free-ebooks/angular-testing-succinctly) - Joseph D. Booth (HTML, PDF, EPUB) (email address required for EPUB and PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1167]:
```suggestion
* [Angular Testing Succinctly](https://www.syncfusion.com/succinctly-free-ebooks/angular-testing-succinctly) - Joseph D. Booth (HTML)
```
This is what we've done for other succinctly titles.

**Pull request merged**